Commonality Analysis Code_2.Rmd

```diff
@@ -0,0 +1,373 @@
+
+cope_ca_data_imputed <- cope_ca_data_imputed[complete.cases(cope_ca_data_imputed$b_cdi_sum,
+                                                           cope_ca_data_imputed$b_gad_mean,
+                                                           cope_ca_data_imputed$b_drs_month,
+                                                           cope_ca_data_imputed$b_cts_rs_mean), ]
+
+# Filtering cope_ca_data_nonimputed
+
+cope_ca_data_nonimputed <- cope_ca_data_nonimputed[complete.cases(cope_ca_data_nonimputed$b_cdi_sum,
+                                                                 cope_ca_data_nonimputed$b_gad_mean,
+                                                                 cope_ca_data_nonimputed$b_drs_month,
+                                                                 cope_ca_data_nonimputed$b_cts_rs_mean), ]
+
+# Creating residual change scores for depression (pre-SSI to 3M) in both the imputed and non-imputed datasets
+
+res_model_depression_imputed <- lm(f1_cdi_sum ~ b_cdi_sum, data = cope_ca_data_imputed)
+cope_ca_data_imputed$residual_change_depression <- res_model_depression_imputed$residuals
+
+res_model_depression_nonimputed <- lm(f1_cdi_mean ~ b_cdi_mean, data = cope_ca_data_nonimputed)
+cope_ca_data_nonimputed$residual_change_depression <- res_model_depression_nonimputed$residuals
+
+# Creating residual change scores for anxiety (pre-SSI to 3M) in both the imputed and non-imputed datasets
+
+res_model_anxiety_imputed <- lm(f1_gad_mean ~ b_gad_mean, data = cope_ca_data_imputed)
+cope_ca_data_imputed$residual_change_anxiety <- res_model_anxiety_imputed$residuals
+
+res_model_anxiety_nonimputed <- lm(f1_gad_mean ~ b_gad_mean, data = cope_ca_data_nonimputed)
+cope_ca_data_nonimputed$residual_change_anxiety <- res_model_anxiety_nonimputed$residuals
+
+# Creating residual change scores for restrictive eating (pre-SSI to 3M) in both the imputed and non-imputed datasets
+
+res_model_drs_imputed <- lm(f1_drs_month ~ b_drs_month, data = cope_ca_data_imputed)
+cope_ca_data_imputed$residual_change_drs <- res_model_drs_imputed$residuals
+
+res_model_drs_nonimputed <- lm(f1_drs_month ~ b_drs_month, data = cope_ca_data_nonimputed)
+cope_ca_data_nonimputed$residual_change_drs <- res_model_drs_nonimputed$residuals
+
+# Creating residual change scores for COVID-19 trauma (pre-SSI to 3M) in both the imputed and non-imputed datasets
+
+res_model_cts_imputed <- lm(f1_cts_rs_mean ~ b_cts_rs_mean, data = cope_ca_data_imputed)
+cope_ca_data_imputed$residual_change_cts <- res_model_cts_imputed$residuals
+
+res_model_cts_nonimputed <- lm(f1_cts_rs_mean ~ b_cts_rs_mean, data = cope_ca_data_nonimputed)
+cope_ca_data_nonimputed$residual_change_cts <- res_model_cts_nonimputed$residuals
+
+```
+
+## Independent Variables

```

Comment:
No need to create residualized change scores, or any change scores for that matter, for the IVs! We are just predicting change based on IV at baseline